harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -241,24 +268,107 @@ def _get_temporal_output_dimension(self,
         return self._get_output_dimension(dimension_name, all_input_values,
                                           output_dimension_units)
 
-    @staticmethod
-    def _get_output_dimension(dimension_name: str,
+    def _get_output_dimension(self, dimension_name: str,
                               input_dimension_values: np.ndarray,
                               dimension_units: str) -> DimensionInformation:
         """ Use `get_resolution` to determine the greatest common divisor of
             all dimension input values. Calculate the output dimension values
             using this resolution, the minimum value of the input dimensions
-            and maximum value of the input dimensions. Finally return a new
-            `DimensionInformation` object encapsulating an aggregated output
-            dimension, which uses a regularly spaced grid that extends to
-            include all input dimension values.
+            and maximum value of the input dimensions.
+
+            Next  use the `_get_dimension_bounds` class method to check if the
+            input dimensions have an associated bounds variable, if so derive
+            the bounds values for the output dimension variable.
+
+            Finally return a new `DimensionInformation` object encapsulating an
+            aggregated output dimension, which uses a regularly spaced grid
+            that extends to include all input dimension values.
 
         """
         grid_resolution = get_resolution(input_dimension_values)
         output_dimension_values = get_grid_values(input_dimension_values,
                                                   grid_resolution)
+
+        bounds_path, bounds_values = self._get_dimension_bounds(
+            dimension_name, output_dimension_values
+        )
+
         return DimensionInformation(dimension_name, output_dimension_values,
-                                    dimension_units)
+                                    dimension_units, bounds_path,
+                                    bounds_values)
+
+    def _get_dimension_bounds(self, dimension_name,
+                              output_dimension_values) -> BoundsTuple:
```

Comment:
This is the meat of the change for DAS-1400. It's not the prettiest method (I was a bit stymied by line length and long variable names), but hopefully the comments throughout make it intelligible.